File: frontend/src/services/dashboardService.js
import api from './api';

/**
 * Dashboard service — fetches data for all dashboard sections.
 *
 * Each function maps to one section of the dashboard UI and
 * one API endpoint. When backend modules are ready (Commits 03–07),
 * only the service impl on the backend changes; these calls stay the same.
 */
const dashboardService = {
  /** GET /api/dashboard/stats — headline stat cards */
  getStats: () => api.get('/dashboard/stats'),

  /** GET /api/dashboard/appointments/today — today's appointments table */
  getTodayAppointments: () => api.get('/dashboard/appointments/today'),

  /** GET /api/dashboard/appointments/upcoming — upcoming widget */
  getUpcomingAppointments: () => api.get('/dashboard/appointments/upcoming'),

  /** GET /api/dashboard/chart/weekly — bar chart data */
  getWeeklyChart: () => api.get('/dashboard/chart/weekly'),
};

export default dashboardService;

File: backend/src/main/java/com/sunrisedental/service/DashboardService.java
package com.sunrisedental.service;

import com.sunrisedental.dto.AppointmentSummaryDto;
import com.sunrisedental.dto.DashboardStatsDto;
import com.sunrisedental.dto.WeeklyChartDto;

import java.util.List;

/**
 * Dashboard data service contract.
 * Each method maps to one section of the dashboard UI.
 */
public interface DashboardService {

    /** Headline statistics displayed in the 4 stat cards. */
    DashboardStatsDto getStats();

    /** Appointments list for the Today's Appointments table. */
    List<AppointmentSummaryDto> getTodayAppointments();

    /** Short list of upcoming appointments for the sidebar widget. */
    List<AppointmentSummaryDto> getUpcomingAppointments();

    /** Weekly appointment counts for the bar chart. */
    WeeklyChartDto getWeeklyChart();
}
